gtsfm/densify/mvsnets/methods/PatchmatchNet/eval_gtsfm.py

```diff
@@ -0,0 +1,435 @@
+                ref_depth_est.astype(np.float32),
+            )
+            save_depth_img(
+                os.path.join(out_folder, "depth_img/conf_{:0>8}.png".format(ref_view)), confidence.astype(np.float32)
+            )
+
+        photo_mask = confidence > photo_thres
+
+        all_srcview_depth_ests = []
+
+        # compute the geometric mask
+        geo_mask_sum = 0
+        for src_view in src_views:
+            # camera parameters of the source view
+            src_intrinsics, src_extrinsics = data["cameras"][src_view]
+
+            # the estimated depth of the source view
+            src_depth_est = depth_list[src_view][0]
+
+            geo_mask, depth_reprojected, x2d_src, y2d_src = check_geometric_consistency(
+                ref_depth_est,
+                ref_intrinsics,
+                ref_extrinsics,
+                src_depth_est,
+                src_intrinsics,
+                src_extrinsics,
+                geo_pixel_thres,
+                geo_depth_thres,
+            )
+            geo_mask_sum += geo_mask.astype(np.int32)
+            all_srcview_depth_ests.append(depth_reprojected)
+
+        depth_est_averaged = (sum(all_srcview_depth_ests) + ref_depth_est) / (geo_mask_sum + 1)
+        # at least 3 source views matched
+        # large threshold, high accuracy, low completeness
+        geo_mask = geo_mask_sum >= 3
+        final_mask = np.logical_and(photo_mask, geo_mask)
+
+        if save_output:
+            os.makedirs(os.path.join(out_folder, "mask"), exist_ok=True)
+            save_mask(os.path.join(out_folder, "mask/{:0>8}_photo.png".format(ref_view)), photo_mask)
+            save_mask(os.path.join(out_folder, "mask/{:0>8}_geo.png".format(ref_view)), geo_mask)
+            save_mask(os.path.join(out_folder, "mask/{:0>8}_final.png".format(ref_view)), final_mask)
+
+        logger.info(
+            "[Densify::PatchMatchNet] processing view:{:0>2}, geo_mask:{:3f} photo_mask:{:3f} final_mask:{:3f} ".format(
+                ref_view, geo_mask.mean(), photo_mask.mean(), final_mask.mean()
+            )
+        )
+
+        height, width = depth_est_averaged.shape[:2]
+        x, y = np.meshgrid(np.arange(0, width), np.arange(0, height))
+
+        valid_points = final_mask
+        x, y, depth = x[valid_points], y[valid_points], depth_est_averaged[valid_points]
+
+        color = ref_img[valid_points]
+        xyz_ref = np.matmul(np.linalg.inv(ref_intrinsics), np.vstack((x, y, np.ones_like(x))) * depth)
+        xyz_world = np.matmul(np.linalg.inv(ref_extrinsics), np.vstack((xyz_ref, np.ones_like(x))))[:3]
+        vertexs.append(xyz_world.transpose((1, 0)))
+        vertex_colors.append((color * 255).astype(np.uint8))
+
+    vertexs_raw = np.concatenate(vertexs, axis=0)
+
+    if save_output:
+        vertex_colors = np.concatenate(vertex_colors, axis=0)
+        vertexs = np.array([tuple(v) for v in vertexs_raw], dtype=[("x", "f4"), ("y", "f4"), ("z", "f4")])
```

Comment:
should be `vertices`, not `vertexs =`

Reply:
I did not check the source codes from original repository carefully.. I'll review them